src/test/java/com/nimbusds/jose/JWEObjectJSONTest.java

```diff
@@ -0,0 +1,354 @@
+
+            new CycleTest(JWEAlgorithm.ECDH_1PU_A192KW, Curve.X25519, EncryptionMethod.A128CBC_HS256),
+            new CycleTest(JWEAlgorithm.ECDH_1PU_A192KW, Curve.X25519, EncryptionMethod.A192CBC_HS384),
+            new CycleTest(JWEAlgorithm.ECDH_1PU_A192KW, Curve.X25519, EncryptionMethod.A256CBC_HS512),
+
+            new CycleTest(JWEAlgorithm.ECDH_1PU_A256KW, Curve.X25519, EncryptionMethod.A128CBC_HS256),
+            new CycleTest(JWEAlgorithm.ECDH_1PU_A256KW, Curve.X25519, EncryptionMethod.A192CBC_HS384),
+            new CycleTest(JWEAlgorithm.ECDH_1PU_A256KW, Curve.X25519, EncryptionMethod.A256CBC_HS512)
+    };
+
+    private static final CycleTest[]  forbiddenECDH_1PUX25519Cycles = new CycleTest[]{
+            new CycleTest(JWEAlgorithm.ECDH_1PU_A128KW, Curve.X25519, EncryptionMethod.A128GCM),
+            new CycleTest(JWEAlgorithm.ECDH_1PU_A128KW, Curve.X25519, EncryptionMethod.A192GCM),
+            new CycleTest(JWEAlgorithm.ECDH_1PU_A128KW, Curve.X25519, EncryptionMethod.A256GCM),
+            new CycleTest(JWEAlgorithm.ECDH_1PU_A128KW, Curve.X25519, EncryptionMethod.XC20P),
+
+            new CycleTest(JWEAlgorithm.ECDH_1PU_A192KW, Curve.X25519, EncryptionMethod.A128GCM),
+            new CycleTest(JWEAlgorithm.ECDH_1PU_A192KW, Curve.X25519, EncryptionMethod.A192GCM),
+            new CycleTest(JWEAlgorithm.ECDH_1PU_A192KW, Curve.X25519, EncryptionMethod.A256GCM),
+            new CycleTest(JWEAlgorithm.ECDH_1PU_A192KW, Curve.X25519, EncryptionMethod.XC20P),
+
+            new CycleTest(JWEAlgorithm.ECDH_1PU_A256KW, Curve.X25519, EncryptionMethod.A128GCM),
+            new CycleTest(JWEAlgorithm.ECDH_1PU_A256KW, Curve.X25519, EncryptionMethod.A192GCM),
+            new CycleTest(JWEAlgorithm.ECDH_1PU_A256KW, Curve.X25519, EncryptionMethod.A256GCM),
+            new CycleTest(JWEAlgorithm.ECDH_1PU_A256KW, Curve.X25519, EncryptionMethod.XC20P),
+    };
+    
+    public void testECDH_1PU_multi_encrypt_decrypt() throws Exception {
+        JWEHeader header = new JWEHeader.Builder(JWEAlgorithm.ECDH_1PU_A256KW, EncryptionMethod.A256CBC_HS512)
+                .agreementPartyVInfo(Base64URL.encode("Alice"))
+                .agreementPartyUInfo(Base64URL.encode("Bob"))
+                .build();
+
+        ECKey aliceKey = generateEC(Curve.P_521, "alice");
+        ECKey bobKey = generateEC(Curve.P_521, "bob");
+        ECKey charlieKey = generateEC(Curve.P_521, "charlie");
+
+        JWEObjectJSON jwe = new JWEObjectJSON(header, new Payload("Hello, world"));
+        ECDH1PUEncrypterMulti encrypterMulti = new ECDH1PUEncrypterMulti(aliceKey, new ECKey[]{ bobKey, charlieKey });
+        jwe.encrypt(encrypterMulti);
+
+        String json = jwe.serialize();
+
+        JWEObjectJSON decrypted = JWEObjectJSON.parse(json);
+        ECDH1PUDecrypterMulti decryptor = new ECDH1PUDecrypterMulti(aliceKey, new ECKey[]{ bobKey, charlieKey });
```

Comment:
How can I pass recipient key `kid`? Can we have a test where we get recipient `kid`s from parsed JSON and pass the corresponding kid:key pairs to decrypt?